graylog2-web-interface/src/components/event-notifications/event-notification-types/HttpNotificationForm.jsx

```diff
@@ -53,16 +100,101 @@ class HttpNotificationForm extends React.Component {
     setIsSubmitEnabled(validationState !== 'error');
   };
 
+  resetSecret = (attribute) => {
+    const { reset } = this.state;
+    reset[attribute] = true;
+    this.setState({ reset });
+
+    this.propagateChange(attribute, { delete_value: true });
+    this.setState({ [attribute]: '' });
+  };
+
+  undoResetSecret = (attribute) => {
+    const { reset } = this.state;
+    reset[attribute] = false;
+    this.setState({ reset });
+
+    this.propagateChange(attribute, { keep_value: true });
+    this.setState({ [attribute]: '******' });
+  };
+
   render() {
     const { config, validation } = this.props;
+    const { api_secret, basic_auth } = config;
+    const { reset } = this.state;
 
     return (
-      <URLWhiteListInput label="URL"
-                         onChange={this.handleChange}
-                         validationState={validation.errors.url ? 'error' : null}
-                         validationMessage={lodash.get(validation, 'errors.url[0]', 'The URL to POST to when an Event occurs.')}
-                         onValidationChange={this.onValidationChange}
-                         url={config.url} />
+      <>
+        <URLWhiteListInput label="URL"
+                           onChange={this.handleChange}
+                           validationState={validation.errors.url ? 'error' : null}
+                           validationMessage={lodash.get(validation, 'errors.url[0]', 'The URL to POST to when an Event occurs.')}
+                           onValidationChange={this.onValidationChange}
+                           url={config.url}
+                           autofocus={false} />
+        <Row>
+          <Col md={12}>
+            {basic_auth?.keep_value ? (
+              <>
+                <ControlLabel>Basic Authentication</ControlLabel>
+                <StyledButton bsStyle="danger" type="button" onClick={() => { this.resetSecret('basic_auth'); }}>
```

Comment:
I would set the `bsStyle` to `default`, to match with what we do in other parts of the product.

Reply:
![Screenshot 2022-04-29 at 12 00 09](https://user-images.githubusercontent.com/92227/165924214-245d0ca1-a5d4-45be-9ee4-569116846b15.png)

Updated.